util/types/datum.go

```diff
@@ -545,7 +545,7 @@ func (d *Datum) compareString(sc *variable.StatementContext, s string) (int, err
 		return CompareString(d.GetString(), s), nil
 	case KindMysqlDecimal:
 		dec := new(MyDecimal)
-		err := dec.FromString([]byte(s))
+		err := sc.HandleTruncate(dec.FromString(hack.Slice(s)))
```

Comment:
ok, what I wanna to say is that
should we add case in TestConvertType for `line1328` ? since it seems hasn't been tested